plugin/unmarshal/unmarshal.go

```diff
@@ -747,56 +789,50 @@ func (p *unmarshal) field(file *generator.FileDescriptor, msg *generator.Descrip
 				valuegoTyp = valuegoAliasTyp
 			}
 
-			p.P(`var keykey uint64`)
-			p.decodeVarint("keykey", "uint64")
-			p.mapField("mapkey", false, m.KeyAliasField)
 			p.P(`if m.`, fieldname, ` == nil {`)
 			p.In()
 			p.P(`m.`, fieldname, ` = make(`, m.GoType, `)`)
 			p.Out()
 			p.P(`}`)
+
+			p.declareMapField("mapkey", false, m.KeyAliasField)
+			p.declareMapField("mapvalue", gogoproto.IsCustomType(field), m.ValueAliasField)
+			p.P(`for iNdEx < postIndex {`)
+			p.In()
+
+			p.P(`var wire uint64`)
+			p.decodeVarint("wire", "uint64")
+			p.P(`fieldNum := int32(wire >> 3)`)
+
+			p.P(`if fieldNum == 1 {`)
+			p.In()
+			p.mapField("mapkey", false, m.KeyAliasField)
+			p.Out()
+			p.P(`} else {`)
```

Comment:
else if fieldNum == 2
else // hmmm.  I guess ignore, since we have to stay compatible with a map object which was manually defined and had extra fields.